gnosis/safe/safe_tx.py

```diff
@@ -415,22 +415,19 @@ def sign(self, private_key: str) -> bytes:
 
         # Insert signature sorted
         if account.address not in self.signers:
-            new_owners = self.signers + [account.address]
-            new_owner_pos = sorted(new_owners, key=lambda x: int(x, 16)).index(
-                account.address
-            )
-            self.signatures = (
-                self.signatures[: 65 * new_owner_pos]
-                + signature
-                + self.signatures[65 * new_owner_pos :]
+            new_tx_signatures = SafeSignature.parse_signature(
+                self.signatures + signature, self.safe_tx_hash
             )
+            self.signatures = SafeSignature.export_signatures(new_tx_signatures)
+
         return signature
 
     def unsign(self, address: str) -> bool:
-        for pos, signer in enumerate(self.signers):
-            if signer == address:
-                self.signatures = self.signatures.replace(
-                    self.signatures[pos * 65 : pos * 65 + 65], b""
-                )
-                return True
+        tx_signatures = SafeSignature.parse_signature(
+            self.signatures, self.safe_tx_hash
+        )
+        new_tx_signatures = list(filter(lambda x: x.owner != address, tx_signatures))
```

Comment:
I would expect one signature by owner, why returns a list? 
Anyway, do you think that make sense change the `unsign`? in this case remove one signature from the middle does not affect the sort.

Reply:
I think the problem is in the name of the method. The `signature` stores the signatures of all owners in bytes... that's why you need to parse them to get the list to manage it.

I did it to keep the logic at the same point. In this case `SafeSignature`. If you see that it doesn't help, I can remove this change, what do you say?

Reply:
If I keep it, I must change the name of the variable to be consistent with the above.

Reply:
> If I keep it, I must change the name of the variable to be consistent with the above.

Done here https://github.com/safe-global/safe-eth-py/pull/1060/commits/314d30b6b43cd30c8880d9573fd8b6819888b5a2